src/index.js

```diff
@@ -0,0 +1,138 @@
+// import "./style.css";
+
+let tasks = [];
+let savedTasks;
+let deleteTaskFn;
+let editTaskFn;
+let saveTaskDescriptionFn;
+
+const createTaskListItem = (task) => {
+    const listItem = document.createElement("div");
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkbox.className = "checkbox";
+    const description = document.createElement("input");
+    description.type = "text";
+    description.className = "input-text";
+    description.value = task.description;
+    description.readOnly = !task.editMode;
+    checkbox.checked = task.completed;
+    checkbox.addEventListener("change", () => {
+        listItem.classList.toggle("completed");
+        listItem.classList.toggle("incomplete");
+        task.completed = !task.completed;
+        description.classList.toggle("cancel-line");
+        savedTasks();
+    });
+
+    const deleteButton = document.createElement("button");
+    deleteButton.textContent = "Delete";
+    deleteButton.addEventListener("click", () => {
+        deleteTaskFn(task.index);
+        savedTasks();
+    });
+
+    const editButton = document.createElement("button");
+    editButton.textContent = task.editMode ? "Save" : "Edit";
+    editButton.addEventListener("click", () => {
+        if (task.editMode) {
+            saveTaskDescriptionFn(task, description.value);
+        } else {
+            editTaskFn(task);
+        }
+    });
+
+    listItem.appendChild(checkbox);
+    listItem.appendChild(description);
+    listItem.appendChild(editButton);
+    listItem.appendChild(deleteButton);
+    listItem.classList.add(task.completed ? "completed" : "incomplete");
+    if (task.completed) {
+        description.classList.add("cancel-line");
+    }
+
+    return listItem;
+};
+
+const renderTaskList = () => {
+    const taskList = document.getElementById("task-list");
+    taskList.innerHTML = "";
+    tasks.sort((a, b) => a.index - b.index);
+    tasks.forEach((task) => {
+        const listItem = createTaskListItem(task);
+        taskList.appendChild(listItem);
+    });
+};
+
+savedTasks = () => {
+    localStorage.setItem("tasks", JSON.stringify(tasks));
+};
+
+const loadTasksFn = () => {
```

Comment:
This function is named using the ```Fn``` suffix which is not consistent with the naming convention you've used in the other functions.

Reply:
thank you for the review. just made the required changes.